06.bowling_object/frame.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require './shot'
+
+class Frame
+  attr_reader :shots
+
+  def initialize(shots, frame_number)
+    @shots = shots
+    @frame_number = frame_number
+  end
+
+  def score(frames)
+    frame_score = @shots.sum(&:point)
+    return frame_score if frame_score != 10 || @frame_number == 9
+
+    next_frame = frames[@frame_number + 1]
+    after_next_frame = frames[@frame_number + 2]
+    bonus_score = if @shots[0].strike?
+                    @frame_number == 8 ? next_frame.shots[0, 2].sum(&:point) : calc_strike(next_frame, after_next_frame)
+                  else
+                    calc_spare(next_frame.shots)
+                  end
```

Comment:
```suggestion
    bonus_score = @shots[0].strike? ? calc_strike(next_frame, after_next_frame) : calc_spare(next_frame.shots)
```


9フレーム目かどうかの判定も calc_strike の中でやってしまえば、ここがシンプルになります

Reply:
calc_strikeで9フレーム目の処理を入れました